### ISSUE TITLE
How to perform scratch training for segmentation model?

### ISSUE THREAD

**Issue by saumyaborwankar** [pyannote/pyannote-audio#927]
According to the tutorial, you can finetune the segmentation model. When I did this for more than 4 epochs the performance starts to deteriorate. So I wanted to train it from scratch. Please guide me thanks.
I am using `pretrained = Model.from_pretrained("pyannote/segmentation")`

**Comment by sty1992928** [comment]
`from pyannote.audio.models.segmentation import PyanNet
  from pyannote.audio.tasks import Segmentation
  import pytorch_lightning as pl

    seg_task = Segmentation(dataset, duration=5, max_num_speakers=4)
    model = PyanNet(task=seg_task, sincnet={'stride': 10})
    trainer = pl.Trainer(max_epochs=1)
    trainer.fit(model)`

**Comment by saumyaborwankar** [comment]
Okay so I did this but now when I use the inference score im getting the raw scores how do i convert them to annotations? 
```
def test(model, protocol, subset="test"):
    from pyannote.audio.utils.signal import binarize
    from pyannote.audio.utils.metric import DiscreteDiarizationErrorRate
    from pyannote.audio.pipelines.utils import get_devices

    (device,) = get_devices(needs=1)
    metric = DiscreteDiarizationErrorRate()
    files = list(getattr(protocol, subset)())

    inference = Inference(model, device=device)
    
    for file in files:
        print(file['uri'])
        reference = file["annotation"]
        h=inference(file)
        hypothesis = binarize(inference(file))
        uem = file["annotated"]
        _ = metric(reference, hypothesis, uem=uem)
```
I have been using this code. So the hypothesis is just a bunch of arrray with 0's

**Comment by hbredin** [comment]
Pipelines are in charge of converting raw scores to annotations.

For example:

```python
from pyannote.audio.pipelines import VoiceActivityDetection
from pyannote.core import Annotation
pipeline = VoiceActivityDetection(segmentation=model)
pipeline.instantiate({"onset": 0.5, "offset": 0.5, "min_duration_on": 0.1, "min_duration_off": 0.1})
hypothesis = pipeline(file)
assert isinstance(hypothesis, Annotation)
```

See https://github.com/pyannote/pyannote-audio/blob/develop/pyannote/audio/pipelines/__init__.py for a list of available pipelines

**Comment by saumyaborwankar** [comment]
But when I have raw scores for segmentation should I use the Voice Activity detection pipeline or some other? Will the voice activity detection pipeline handle different labels given for segmentation when doing inference.

**Comment by saumyaborwankar** [comment]
I have another issue so after I trained the segmentation model it says that the local DER is 39 but when I tried the same model with the pipeline the DER jumped to 120 any reason why this is an issue?

**Comment by hbredin** [comment]
This is because the `SpeakerSegmentation` pipeline does not do global speaker diarization. 

If you are interested in speaker diarization of the whole audio recording, you have to use (and tune) the `SpeakerDiarization` pipeline.

**Comment by saumyaborwankar** [comment]
> This is because the `SpeakerSegmentation` pipeline does not do global speaker diarization.
> 
> If you are interested in speaker diarization of the whole audio recording, you have to use (and tune) the `SpeakerDiarization` pipeline.

Could you let me know the steps if you have the time.

**Comment by saumyaborwankar** [comment]
Any update on this?

**Comment by sty1992928** [comment]
Did you want to use the speaker segmentation to imporve the performance of speaker diarization?
See the Resegmentaion Example in https://huggingface.co/pyannote/segmentation

**Comment by saumyaborwankar** [comment]
My goal is speaker diarization, for that the speaker diarization pipeline uses segmentation and embedding models. I have trained the segmentation model alone and will be using the pretrained embedding model. Now my question is how do i train the speaker diarization pipeline. Because my segmentation DER is 39 but when i use the segmentation model with speajer diarization pipeline for global diarization on a big file it, the DER becomes 126

**Comment by sty1992928** [comment]
What's the Der when you only use the speaker diarization pipeline?

**Comment by saumyaborwankar** [comment]
I want to train speaker diarization now I saw that the speaker diarization pipeline uses segmentation and embedding modules so I trained segmentation model and i used the pretrained embedding model. Now my segmentation model alone is giving 39DER i dont know how and when i use it with speaker diarization pipeline it gives 126 DER. Using speaker diarization pipeline alone gives me DER of 234.
Please give me steps on how to train whole speaker diarization pipeline so I can use it on 20 minutes long audio clip which has 8 speakers

**Comment by kaleaniket** [comment]
> I want to train speaker diarization now I saw that the speaker diarization pipeline uses segmentation and embedding modules so I trained segmentation model and i used the pretrained embedding model. Now my segmentation model alone is giving 39DER i dont know how and when i use it with speaker diarization pipeline it gives 126 DER. Using speaker diarization pipeline alone gives me DER of 234. Please give me steps on how to train whole speaker diarization pipeline so I can use it on 20 minutes long audio clip which has 8 speakers

Hello,
Did you find anything on training a full speaker diarization model?

**Comment by hbredin** [comment]
@kaleaniket, see [this tutorial](https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/adapting_pretrained_pipeline.ipynb)
